pagopa/io-wallet
# chore: [SIW-1209] Use base64url for iOS integrity keys

#### List of Changes
- Change the encoding format of the public key for iOS from `base64` to `base64url`.



#### Motivation and Context



Currently, `io-wallet` might return a 500 while creating a wallet instance on iOS due to the hardware key tag being encoded in base64 and thus having characters which can't be stored on the underlying Cosmos DB.
[This PR](https://github.com/pagopa/io-react-native-wallet/pull/88) converts the base64 key tag to base64url in the `io-react-native-wallet`, thus making it safe to be stored on Cosmos DB.
This doesn't happen on Android as the we can decide the name of the key tag used.
Changing the encoding format when verifying the public key solves the issue.

#### How Has This Been Tested?

- Run an instance of `apps/io-wallet-user-func` from [io-wallet](https://github.com/pagopa/io-wallet) with an appropriate `local.settings.json` (I can provide it privately);
- Run an instance of `dev-proxy` in this example folder by providing a `.env` file pointing at the running azure function of the previous point (`http://localhost:7071/api/v1` by default). I can provide the remaining variables privately);
- Run the example app on both Android and iOS by pointing at the branch used for [this PR](https://github.com/pagopa/io-react-native-wallet/pull/88/) and by providing a `.env` file. The `WALLET_PROVIDER_BASE_URL` should point to the IP of your local machine URL where the `dev-proxy` is running (again, I can provide the remaining variables privately). Consequentially press every available button in the app and everything should work just fine.

#### Checklist:
